RecoPixelVertexing/PixelTriplets/plugins/PixelQuadrupletGenerator.h

```diff
@@ -0,0 +1,38 @@
+#include "RecoTracker/TkHitPairs/interface/HitPairGenerator.h"
+#include "RecoPixelVertexing/PixelTriplets/interface/HitTripletGeneratorFromPairAndLayers.h"
+#include "CombinedHitQuadrupletGenerator.h"
+#include "RecoTracker/TkSeedingLayers/interface/SeedingLayer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoPixelVertexing/PixelTriplets/interface/HitQuadrupletGeneratorFromTripletAndLayers.h"
+
+#include <utility>
+#include <vector>
+
+class SeedComparitor;
+
+class PixelQuadrupletGenerator : public HitQuadrupletGeneratorFromTripletAndLayers {
+
+typedef CombinedHitQuadrupletGenerator::LayerCacheType       LayerCacheType;
+
+public:
+  PixelQuadrupletGenerator( const edm::ParameterSet& cfg, edm::ConsumesCollector& iC);
+
+  virtual ~PixelQuadrupletGenerator();
+
+  virtual void hitQuadruplets( const TrackingRegion& region, OrderedHitSeeds& result,
+                               const edm::Event & ev, const edm::EventSetup& es,
+                               SeedingLayerSetsHits::SeedingLayerSet tripletLayers,
+                               const std::vector<SeedingLayerSetsHits::SeedingLayer>& fourthLayers) override;
```

Comment:
The `SeedingLayerSetsHits::SeedingLayerSet` is a pointer and two iterators (to `std::vector`), and its interface is const. At the time it was a conscious decision from me to pass it by value in triplet generators (and hence also here), but perhaps the interface of the triplet and quadruplet generators would be more uniform if also this parameter would be passed by const reference (anyway these functions are not called from tight loops). With 100 events IgProf shows no difference in time.

@VinInn If you don't disagree, I'll change the passing to const reference (also for triplet generators).